frontend/src/components/NewRunParameters.tsx

```diff
@@ -16,18 +16,48 @@
 
 import * as React from 'react';
 import { commonCss } from '../Css';
+import Button from '@material-ui/core/Button';
+import InputAdornment from '@material-ui/core/InputAdornment';
 import TextField from '@material-ui/core/TextField';
 import { ApiParameter } from '../apis/pipeline';
+import { stylesheet } from 'typestyle';
+import { color, spacing } from '../Css';
+import Editor from './Editor';
 
 export interface NewRunParametersProps {
   initialParams: ApiParameter[];
   titleMessage: string;
   handleParamChange: (index: number, value: string) => void;
 }
 
-class NewRunParameters extends React.Component<NewRunParametersProps> {
+export interface NewRunParametersState {
+  isBeingEdited: { [key: number]: boolean };
+}
+
+const css = stylesheet({
+  key: {
+    color: color.strong,
+    flex: '0 0 50%',
+    fontWeight: 'bold',
+    maxWidth: 300,
+  },
+  nonEditableInput: {
+    color: color.secondaryText,
+  },
+  row: {
+    borderBottom: `1px solid ${color.divider}`,
+    display: 'flex',
+    padding: `${spacing.units(-5)}px ${spacing.units(-6)}px`,
+  },
+});
+
+class NewRunParameters extends React.Component<NewRunParametersProps, NewRunParametersState> {
   constructor(props: any) {
     super(props);
+
+    this.state = {
+      isBeingEdited: {},
```

Comment:
An early suggestion is: instead of keeping a dictionary of state in `NewRunParameters` component, state will be easier to manage if you define a sub component named maybe `EnhancedTextField`, this component keeps a boolean state of whether editor mode is turned on and the extra logic of switching between an editor and a text field.

The extra layer of abstraction would make both `NewRunParameters` component and the `EnhancedTextField` component easier to read.

Since the new component is only used here, it's fine to keep it in the same file and there's no need to make it very general.

Reply:
After the change, you may find there are too many snapshot changes... that's a problem of our unit tests (they test too much implementation details). We are moving towards testing components more by `mount` instead of `shallow`.

Reply:
I have reformatted it in the requested way and updated the snapshots